domain/pricing.go

```diff
@@ -32,6 +33,9 @@ type PricingSource interface {
 	// InitializeCache initialize the cache for the pricing source to a given value.
 	// Panics if cache is already set.
 	InitializeCache(*cache.Cache)
+
+	// ShouldFallback determins what pricing source should be fallen back to in case this pricing source fails
+	ShouldFallback(quoteDenom string) PricingSourceType
```

Comment:
nit: would it make sense naming this as `GetFallBackStrategy` for readability? I would expect something that starts with `Should` to output a boolean